server.js

```diff
@@ -3,36 +3,67 @@ import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import booksData from './data/books.json'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+const Book = new mongoose.model('Book', {
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: String,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+})
+
+if (process.env.RESET_DATABASE) {
+  const populateDatabase = async () => {
+    await Book.deleteMany()
+
+    booksData.forEach(item => {
+      const newBook = new Book(item)
+      newBook.save()
+    })
+
+  }
+  populateDatabase()
+}
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(`Hello, Welcome to Sandra's API for books`)
+})
+
+app.get('/books', async (req, res) => {
+  const allBooks = await Book
+    .find({ authors: { "$regex": req.query.author ?? '' }, title: { "$regex": req.query.title ?? '' } })
+    .sort({ title: 'asc', authors: -1 })
```

Comment:
To make this `sort()` do what it supose to do, you need to add `.exec()` after it like this `.find().sort().exec()` 😺